src/App.tsx

```diff
@@ -9,48 +9,48 @@
 import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
-import { getUserPosts } from './api/posts';
+import { useAppDispatch, useAppSelector } from './app/hooks';
+import {
+  fetchUserPosts,
+  clearPosts,
+  setSelectedPost,
+} from './features/posts/posts';
+import { setAuthor } from './features/author/author';
 import { User } from './types/User';
 import { Post } from './types/Post';
 
 export const App: React.FC = () => {
-  const [posts, setPosts] = useState<Post[]>([]);
-  const [loaded, setLoaded] = useState(false);
-  const [hasError, setError] = useState(false);
-
-  const [author, setAuthor] = useState<User | null>(null);
-  const [selectedPost, setSelectedPost] = useState<Post | null>(null);
-
-  function loadUserPosts(userId: number) {
-    setLoaded(false);
+  const author = useAppSelector(state => state.author.author);
+  const { hasError, loaded, posts, selectedPost } = useAppSelector(
+    state => state.posts,
+  );
 
-    getUserPosts(userId)
-      .then(setPosts)
-      .catch(() => setError(true))
-      // We disable the spinner in any case
-      .finally(() => setLoaded(true));
-  }
+  const dispatch = useAppDispatch();
 
   useEffect(() => {
-    // we clear the post when an author is changed
-    // not to confuse the user
-    setSelectedPost(null);
+    dispatch(setSelectedPost(null));
 
     if (author) {
-      loadUserPosts(author.id);
+      dispatch(fetchUserPosts(author.id));
     } else {
-      setPosts([]);
+      dispatch(clearPosts());
     }
   }, [author]);
```

Comment:
Ensure that all dependencies used within the `useEffect` are included in the dependency array. Currently, only `author` is included, which is correct if it's the only dependency. If there are other dependencies, they should be added to avoid potential bugs.